libyul/AsmParser.cpp

```diff
@@ -78,14 +108,61 @@ unique_ptr<Block> Parser::parse(std::shared_ptr<Scanner> const& _scanner, bool _
 	return nullptr;
 }
 
+langutil::Token Parser::advance()
+{
+	auto const token = ParserBase::advance();
+	if (m_useSourceLocationFrom == UseSourceLocationFrom::Comments)
+		fetchSourceLocationFromComment();
+	return token;
+}
+
+void Parser::fetchSourceLocationFromComment()
+{
+	solAssert(m_charStreamMap.has_value(), "");
+
+	if (m_scanner->currentCommentLiteral().empty())
+		return;
+
+	static std::regex const lineRE = std::regex(
+		R"~~~((^|\s+)@src\s+(\d+):(-1|\d+):(-1|\d+)(\s+|$))~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+
+	std::smatch matchResult;
+	if (!std::regex_search(m_scanner->currentCommentLiteral(), matchResult, lineRE))
```

Comment:
We have to find the last match. Can you add a test for
```
/// @src 1:2:3
/// @src 4:5:7
sstore(0, 1)
```
The expected source location should be `4:5:7`